src/plugins/shared-variables/slate/variables-text-content.ts

```diff
@@ -68,36 +88,4 @@ export function getOrFindSharedModel(textContent: TextContentModelType) {
   }
 
   return sharedModel;
-}
-
-export function updateAfterSharedModelChanges(
```

Comment:
I think the unused variable list in the dialogs renders the need for this obsolete -- it seems only to be useful if deleting a variable in a text tile is supposed to delete that variable in all tiles.  If I'm wrong, I *think* the replacement code would look something like this:
```
if (textContent.editor) {
   const variables = getVariables(textContent);
   for (const [node, path] of Editor.nodes(textContent.editor, {at: [], mode: 'all'})) {
     if (Editor.isInline(textContent.editor, node) && isVariableElement(node)) {
       if (!variables.find(v => v.id === node.reference)) {
         Transforms.removeNodes(textContent.editor, { at: path });
       }
     }
   }
 }  
```

Reply:
I think this function would be necessary if there were a way to completely delete variables in another tile or dialog. Then, if that variable existed as a chip in the text tile, we'd need to remove it. So this function would need to make sure that all variables in the text tile are also in the shared model, and if one isn't, delete the chip. Since there isn't a way to permanently remove a variable at this time, we don't need to worry about it for now. Also, this would only be an issue if it's possible to delete a variable that's being used by a tile, which we might disallow (and only allow unused variables to be deleted).

You're right that we don't want to completely delete variables when a variable chip is deleted 👍